### PR TITLE
Update to ~~TAMS~~ AMS framework

### PR DESCRIPTION
**Pull request by jamelvin**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [✓] Feature enhancement

## Description of the pull-request

This is an update to the ~~TAMS~~ AMS framework that allows for stable runs with large CFLs.  This includes
  - Renaming of TAMS to AMS  (Note that SSTTAMSAveragesAlg -> SSTAMSAveragesAlg is being treated as rm/add)
  - Implicit treatment of the Mean Turbulent Stress
  - Coupling of averaging with Picard iterations
  - Minor updates to Forcing stress
  - Minor updates to various limiters and blending 

  - Documentation has been updated

  - As far as I'm aware* it passes all tests except for the following due to changes in the AMS framework:
      - amsChannelEdge
      - SSTAMSOversetRotCylinder
      - SSTAMSChannelEdge

   * I had some difficulty successfully running a handful of tests on Eagle using master and this branch

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [✓] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [✓] Linux
    - [ ] MacOS
  - Compilers 
    - [✓] GCC
    - [ ] LLVM/Clang
    - [✓] Intel compilers
    - [ ] NVIDIA CUDA
- [✓] Compiles without warnings
- [ See Above ] Passes all unit tests
- [ See Above ] Passes all regression tests
- [✓] Documentation builds without errors

*For new code updates*
- [✓] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [ ] New regression tests exercising the new code

### CODE REVIEW

File: include/Enums.h [hunk 1]
```diff
@@ -75,7 +75,7 @@ enum EquationType {
   EQ_MESH_DISPLACEMENT = 7,
   EQ_SPEC_DISS_RATE = 8,
   EQ_MASS_FRACTION = 9,
-  EQ_TAMS = 10,
+  EQ_AMS = 10,
```
Comment (marchdf) [inline include/Enums.h:78]: Do we need this anymore?

File: include/Enums.h [hunk 2]
```diff
@@ -97,7 +97,7 @@ static const std::string EquationTypeMap[] = {
   "MeshVelocity",
   "Specific_Dissipation_Rate",
   "Mass_Fraction",
-  "TAMS",
+  "AMS",
```
Comment (marchdf) [inline include/Enums.h:100]: And this?

**Review by marchdf — APPROVED** [review]

File: include/Enums.h (hunk 1 above)
Comment (jamelvin) [reply include/Enums.h]: nope

File: include/node_kernels/TKESSTNodeKernel.h [hunk 3]
```diff
@@ -48,14 +47,15 @@ class TKESSTNodeKernel : public NGPNodeKernel<TKESSTNodeKernel>
   stk::mesh::NgpField<double> density_;
   stk::mesh::NgpField<double> tvisc_;
   stk::mesh::NgpField<double> dudx_;
+  stk::mesh::NgpField<double> coords_;
```
Comment (jamelvin) [inline include/node_kernels/TKESSTNodeKernel.h:50]: going to remove these, they shouldn't be there as they are from previous debugging

File: include/EigenDecomposition.h [hunk 4]
```diff
@@ -290,21 +298,97 @@ void reconstruct_matrix_from_decomposition(const T (&D)[3][3],
...
+  const auto check_one = disc < 0.0;
+  const bool exit_now = stk::simd::are_all(check_one);
+  if (exit_now) {
+    NaluEnv::self().naluOutputP0()
+      << "Error, complex eigenvalues in EigenDecomposition::general_eigenvalues"
+      << disc << "([[" << A[0][0] << "," << A[0][1] << "," << A[0][2] << "],["
+      << A[1][0] << "," << A[1][1] << "," << A[1][2] << "],[" << A[2][0] << ","
+      << A[2][1] << "," << A[2][2] << "]])" << std::endl;
+    throw std::runtime_error("Exiting... ");
+  }
```
Comment (sayerhs) [inline include/EigenDecomposition.h:352]: In MPI runs, the `exit_now` situation might occur on an mpi rank that is not the root process. In this situation, the code will exit with an obscure "Exiting..." message while the error details are sent to `/dev/null` by `naluOutputP0`. In general I also recommend repeating the general error message 
`Error, complex eigenvalues in EigenDecomposition::general_eigenvalues` in the runtime_error argument also. 

```suggestion
  if (exit_now) {
    NaluEnv::self().naluOutput()
      << "Error, complex eigenvalues in EigenDecomposition::general_eigenvalues"
      << disc << "([[" << A[0][0] << "," << A[0][1] << "," << A[0][2] << "],["
      << A[1][0] << "," << A[1][1] << "," << A[1][2] << "],[" << A[2][0] << ","
      << A[2][1] << "," << A[2][2] << "]])" << std::endl;
    throw std::runtime_error("Exiting... ");
  }
```

File: include/ngp_algorithms/SSTAMSAveragesAlg.h [hunk 5]
```diff
@@ -21,41 +20,49 @@ namespace nalu {
...
   unsigned avgResAdeq_{stk::mesh::InvalidOrdinal};
+  unsigned avgResAdeqN_{stk::mesh::InvalidOrdinal};
   unsigned tvisc_{stk::mesh::InvalidOrdinal};
-  unsigned alpha_{stk::mesh::InvalidOrdinal};
+  unsigned visc_{stk::mesh::InvalidOrdinal};
+  unsigned beta_{stk::mesh::InvalidOrdinal};
   unsigned Mij_{stk::mesh::InvalidOrdinal};
+  unsigned wallDist_{stk::mesh::InvalidOrdinal};
+
+  // FIXME: What to do with beta_kol in SST? This needs some thought...
```
Comment (sayerhs) [inline include/ngp_algorithms/SSTAMSAveragesAlg.h:64]: Is this still a FIXME or has this been addressed?

**Review by sayerhs — APPROVED** [review]

File: include/EigenDecomposition.h (hunk 4 above)
Comment (jamelvin) [reply include/EigenDecomposition.h:352]: Thanks, updated!

**Pull request merged**